### PR TITLE
Add inspection-batch-with-output-gathering example

### PR DESCRIPTION
**Pull request by CermakM**
This workflow submits a Job which creates `n` pod replicas and runs them in a sequence.
Additionally, it mounts an artifact repository to store the resulting artifacts.

NOTE: The artifact repository has to be configured via ConfigMap, see
https://github.com/argoproj/argo/blob/master/ARTIFACT_REPO.md
or use the Argo ansible role: https://github.com/CermakM/ansible-role-argo-workflows

successCondition: `n` pods have finished successfully and are present in the cluster

The workflow simulates the scenario when we want to create a batch of inspections
and store the outputs to the artifact store.

Signed-off-by: Marek Cermak <macermak@redhat.com>

### CODE REVIEW

**Comment by CermakM** [comment]
[Create examples of Argo Workflows.](https://trello.com/c/rbSVX7o4/730-create-examples-of-argo-workflows)

File: examples/inspection-batch-with-output-gathering.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          completions: {{workflow.parameters.batch-size}}
+          template:         
+            metadata:
+              name: "{{workflow.name}}"
+              labels:
+                batch: "{{workflow.parameters.batch-name}}"
+            spec:
+              containers:
+              - name: fake-inspection
+                image: alpine:latest
```
Comment (goern) [inline examples/inspection-batch-with-output-gathering.yaml]: ubi8!

**Review by goern — CHANGES_REQUESTED** [review]

File: examples/inspection-batch-with-output-gathering.yaml (hunk 1 above)
Comment (CermakM) [reply examples/inspection-batch-with-output-gathering.yaml]: Ye ye :D

**Review by goern — APPROVED** [review]

**Comment by goern** [comment]
@CermakM this needs some love

**Comment by CermakM** [comment]
recheck

**Pull request merged**